src/jobs/fetch/executeFetch.ts

```diff
@@ -0,0 +1,226 @@
+ *
+ * Port of source code: https://github.com/fluxprotocol/basic-fetch-wasm
+ *
+ * Sample input arguments:
+ * ```
+ * [
+ *   'FetchJob',
+ *   '[{"source_path":"$.market_data.current_price.usd","end_point":"https://api.coingecko.com/api/v3/coins/ethereum"}]',
+ *   'number',
+ *   '1000000'
+ * ]
+ * ```
+ *
+ * Sample WASM valid output:
+ * ```
+ * {
+ *  code: 0,
+ *  gasUsed: '2154667330',
+ *  logs: [
+ *    'Matching values found: 1',
+ *    'url: https://api.coingecko.com/api/v3/coins/ethereum, source: $.market_data.current_price.usd, result: Some(BigDecimal("2760.01"))',
+ *    'used sources: 1/1',
+ *    '',
+ *    '{"type":"Valid","value":"2760010000"}'
+ *  ]
+ * }
+ * ```
+ *
+ * Sample WASM invalid output:
+ * ```
+ * {
+ *  code: 1,
+ *  logs: [
+ *    "thread 'main' panicked at 'assertion failed: `(left != right)`",
+ *    '  left: `0`,',
+ *    " right: `0`: ERR_NO_SOURCES', src/main.rs:31:5",
+ *    'note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace',
+ *    ''
+ *  ],
+ *  gasUsed: '1487247'
+ * }
+ * ```
+ * @param {ExecuteInput} args
+ */
+export async function executeFetch(args: string[]): Promise<ExecuteResult> {
+    const funcName = "[executeBasicFetch]";
+
+    let logs: string[] = [];
+
+    // Check arguments (not gracefully checked in WASM VM)
+    if (args.length < 3) {
+        return buildExecutionOutput({
+            code: "ERR_INSUFFICIENT_ARGS",
+            message: `${funcName} called with insufficient arguments (args.length < 3)`,
+            args
+        });
+    }
+
+    const sources: Source[] = JSON.parse(args[1]);
+    const sourcesType: string = args[2];
+
+    let stringResult = "";
+    let numberResult = new Big(0);
+    let usedSources = 0;
+
+    // Check - no sources
+    if (sources.length == 0) {
+        return buildExecutionOutput({
+            code: "ERR_NO_SOURCES",
+            message: `${funcName} called with no sources (args[1].length == 0)`,
+            args
+        });
+    }
+
+    // ERRORs - types
+    if (sourcesType == 'string') {
+        if (sources.length != 1) {
+            return buildExecutionOutput({
+                code: "ERR_TOO_MUCH_SOURCES",
+                message: `${funcName} called with too many sources for sourcesType = 'string' (sources.length != 1)`,
+                args
+            });
+        }
+    } else if (sourcesType == 'number') {
+        if (args[3] === 'undefined') {
+            return buildExecutionOutput({
+                code: "ERR_NO_MULTIPLIER",
+                message: `${funcName} called with no multiplier for sourcesType = 'number' (args[3] === 'undefined')`,
+                args
+            });
+        }
+    } else {
+        return buildExecutionOutput({
+            code: "ERR_UNSUPPORTED_TYPE",
+            message: `${funcName} called with unsupported sourcesType (sourceType != 'string' | 'number')`,
+            args
+        });
+    }
+
+    for (let source of sources) {
```

Comment:
for await (let source... 

Otherwise it will not wait on the previous iteration